octoprint_simple_filament_change_buttons/static/js/simple_filament_change_buttons.js

```diff
@@ -11,29 +11,70 @@ $(function() {
 		// assign the injected parameters
 		self.controlViewModel = parameters[0];
 
+		self.unloadFilament = function() {
+			cmds = [
+				// Heat to specified temp
+				"M109 S" + self.temperature(),
+				// Do unload
+				"M702",
+				// Cool hotend
+				"M104 S0"
+			]	
+			OctoPrint.control.sendGcode(cmds)
+		}
+
+		self.loadFilament = function() {
+			cmds = [
+				// Heat to specified temp
+				"M109 S" + self.temperature(),
+				// Do load
+				"M701",
+				// Cool hotend
+				"M104 S0"
+			]	
+			OctoPrint.control.sendGcode(cmds)
+		}
+
+		self.changeFilament = function() {
+			cmds = [
+				// Change filament at specified temp
+				"M600 R" + self.temperature(),
+			]		
+			OctoPrint.control.sendGcode(cmds)
+		}
+
+		self.resume = function() {
+			cmds = [
+				'M108'
+			]		
+			OctoPrint.control.sendGcode(cmds)
+		}
+
+		self.temperature = function()
+		{
+			// The temperature input field has a "placeholder" attribute that is equal to it's name
+			return $('*[placeholder = "Temperature"]')[0].value
```

Comment:
Is this the expected way to retrieve input? I've not done that in a Octoprint plugin before, but it seems odd.

Reply:
It is a but odd, but given that this is a simple custom control plugin, the only way to have an input that is shared across multiple controls (rather than a temperature for each button) is to extract that information from the DOM with (jquery in this case)